File: Web/src/api/userChallenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetUserChallengeList() {
  return apiClient.post("/userChallenge/GetUserChallengeList").then(handleResponse).catch(handleError)
}

function ResetUserChallenge(data) {
  return apiClient.post("/userChallenge/ResetUserChallenge", data).then(handleResponse).catch(handleError)
}

function CheckChallengeFlag(data) {
  return apiClient.post("/userChallenge/CheckChallengeFlag", data).then(handleResponse).catch(handleError)
}

const userChallengeApi = {
  GetUserChallengeList,
  ResetUserChallenge,
  CheckChallengeFlag
}

export default userChallengeApi

File: Server/route/userChallenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserChallengeRoute(r *gin.Engine) {
	userChallenge := r.Group("/userChallenge")
	userChallenge.Use(middleware.JWTAuthMiddleware())
	{
		userChallenge.POST("GetUserChallengeList", api.GetUserChallengeList)
		userChallenge.POST("ResetUserChallenge", api.ResetUserChallenge)
		userChallenge.POST("CheckFlag", api.CheckFlag)
	}
}
